scripts/prepare-unity.sh

```diff
@@ -37,11 +37,22 @@ export GOPATH
 
 if [ -d "$IMPROBABLE_UNITY_VERSIONED" ]; then
   echo "Unity is already installed in:" $IMPROBABLE_UNITY_VERSIONED
-  exit 0
 else
   echo "Could not find a Unity $UNITY_PINNED_VERSION installation."
 fi
 
+if [ -d "$IMPROBABLE_UNITY_VERSIONED/PlaybackEngines/LinuxStandaloneSupport" ] || \
+   [ -d "$IMPROBABLE_UNITY_VERSIONED/PlaybackEngines/AndroidPlayer" ] || \
+   [ -d "$IMPROBABLE_UNITY_VERSIONED/PlaybackEngines/iOSSupport" ] || \
+   [ -d "$IMPROBABLE_UNITY_VERSIONED/PlaybackEngines/WindowsStandaloneSupport" ] || \
+   [ -d "$IMPROBABLE_UNITY_VERSIONED/PlaybackEngines/MacStandaloneSupport" ]
+then
+  echo "All PlaybackEngines are already installed in: $IMPROBABLE_UNITY_VERSIONED/PlaybackEngines."
+  exit 0
+else
+  echo "Could not find all playback engines."
```

Comment:
What happens if we get here because of a missing playback engine and try to reinstall all of Unity?

Reply:
then it re-installs all of Unity with all the modules specified above - did a test run on an agent and it appears to reinstall fine